Sources/ConcurrencyExtras/Result.swift

```diff
@@ -1,10 +1,10 @@
-extension Result where Failure == Swift.Error {
+extension Result where Failure: Swift.Error {
```

Comment:
I think this isn't needed at all.

```suggestion
extension Result {
```